src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -36,8 +36,12 @@ end
 # Mask immersed fields
 function masking_actions!(model)
     η = displacement(model.free_surface)
-    masking_events = Any[mask_immersed_field!(field)
-                         for field in merge(model.auxiliary_fields, prognostic_fields(model)) if field !== η]
+    prog = prognostic_fields(model)
+
+    # selects u, v, if they're prognostic variables
+    prognostic_vels = Tuple( val for (key, val) in zip(keys(prog), prog) if Symbol(key) in (:u, :v) )
+
+    masking_events = Any[mask_immersed_field!(field) for field in prognostic_vels]
```

Comment:
```suggestion
    mask_velocities_event = mask_immersed_velocities!(model.velocities, architecture(model), model.grid)
```

some changes needed below. Also, may need to dispatch on `mask_immersed_velocities!(::PrescribedVelocityFields, args...) = tuple(NoneEvent())` perhaps.